WalletWasabi/WabiSabi/Client/IDestinationProvider.cs

```diff
@@ -1,9 +1,23 @@
 using NBitcoin;
 using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using WalletWasabi.WabiSabi.Backend.Rounds;
+using WalletWasabi.WabiSabi.Models;
 
 namespace WalletWasabi.WabiSabi.Client;
 
 public interface IDestinationProvider
 {
-	IEnumerable<IDestination> GetNextDestinations(int count);
+	Task<IEnumerable<IDestination>> GetNextDestinations(int count);
+	
+	Task<IEnumerable<PendingPayment>> GetPendingPayments(RoundParameters roundParameters,ImmutableArray<AliceClient> registeredAliceClients);
+}
+
+public class PendingPayment
+{
+	public IDestination Destination { get; set; }
+	public Money Value { get; set; }
+	public Action PaymentStarted { get; set; }
+	public Action PaymentFailed { get; set; }
```

Comment:
I'm still playing around on the best way to signal to the wallet that this coinjoin has the intent of batching specific payments and to cancel the intent if the round fails.